colcon_ros/task/ament_python/build.py

```diff
@@ -34,4 +40,58 @@ def __init__(self):  # noqa: D107
             'ament_prefix_path', Path(args.install_base),
             self.context.pkg.name, 'AMENT_PREFIX_PATH', '', mode='prepend')
 
+        # get options from the Python manifest
+        try:
+            env = await get_command_environment(
+                'setup_py', args.build_base, self.context.dependencies)
+        except RuntimeError as e:
+            logger.error(str(e))
+            return 1
+        setup_py_data = get_setup_data(self.context.pkg, env)
+
+        # check if the package index and manifest are being installed
+        data_files = get_data_files_mapping(
+            setup_py_data.get('data_files', []))
+        installs_package_index = False
+        installs_package_manifest = False
+
+        # check if package index and manifest are being installed
+        for source, destination in data_files.items():
+            # work around data files incorrectly defined as not relative
+            if os.path.isabs(source):
+                source = os.path.relpath(source, args.path)
+            if (
+                destination == 'share/ament_index/resource_index/packages/' +
+                self.context.pkg.name
+            ):
+                installs_package_index = True
+            elif (
+                source == 'package.xml' and
+                destination == 'share/{self.context.pkg.name}/package.xml'
+                .format_map(locals())
+            ):
+                installs_package_manifest = True
+
+        # for now implicitly install the marker and the manifest
+        if not installs_package_index:
+            # TODO remove magic helper in the future
+            # create package marker in ament resource index
+            create_file(
```

Comment:
This isn't actually a simple revert of #74, it's a behavioral change that will need to be tested. I would be must faster to approve an actual revert.

Reply:
Can you clarify where exactly you see a problem with the proposed change?

Reply:
Reviewers will need to carefully verify that `installs_package_index` and `installs_package_manifest` are set properly across different `data_files`, when that wouldn't be necessary for a actual revert.

Reply:
Yes, a reviewer needs to read the handful of involved lines. Since that is not a significant task I don't think separating it into another PR is necessary.

Reply:
@dirk-thomas, you're blocking fixing breakage behind a feature you want to add. Reviewing PRs, especially ones this important, is not just a quick glance at the code. It's actually testing it. At least it would be for me.

Reply:
> you're blocking fixing breakage behind a feature you want to add.

No, I am not blocking anything. This PR is waiting for a reviewer which blocks this PR to move forward. If other reviewers think too that this is a too complex change to comprehend then I will reconsider.

Reply:
I find it interesting that you [blame your reviewers for breakage](https://discourse.ros.org/t/can-we-chat-about-breaking-changes-in-stable-releases/10689/15?u=kyrofa), and then don't actually work to make your reviewers comfortable.

Reply:
I won't continue this discussion here. We simply disagree on this.

The time spent on this conversation could have already been used for reviewing / testing this change and we would have been making actual progress.

Reply:
Indeed, we do. I will be happy to review a standalone revert, then we can evaluate this feature you're adding on its own merits. Outside of that possibility, you'll need to find another reviewer.